zerver/lib/user_status.py

```diff
@@ -113,3 +113,22 @@ def update_user_status(
         user_profile_id=user_profile_id,
         defaults=defaults,
     )
+
+
+def get_user_status(user_profile_id: int) -> UserInfoDict:
+    result = (
+        UserStatus.objects.filter(user_profile_id=user_profile_id)
+        .values(
+            "user_profile_id",
+            "user_profile__presence_enabled",
+            "status_text",
+            "emoji_name",
+            "emoji_code",
+            "reaction_type",
+        )
+        .first()
+    )
+
+    if not result:
```

Comment:
Maybe instead of `result`, this could be `status_set_by_user` so this reads more clearly ...

```python
if not status_set_by_user:
    return {}
```